src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -1,171 +1,594 @@
+         * Updates plot when brushing.
+         */
+        function updateChart() {
+            let selectionEvent = d3.event.selection;
+            selectedZoomBtn.style.display = "inline-block";
+            selectedZoomBtn.addEventListener("click",zoomIn,true)
+            document.addEventListener('keydown',zoomInKeyEvent,true);
+            extent = selectionEvent;
+        }
 
-            }
+        /**
+         * Removes brush, keyboard event and zoom button when end event is captured.
+         */
+        const endBrushing = function (_) {
+            brushElem.call(brush.move,null);
+            selectedZoomBtn.style.display = "none";
+            selectedZoomBtn.removeEventListener("click",zoomIn,true)
+            document.removeEventListener('keydown',zoomInKeyEvent,true);
         };
+
+        let endEvents = ['click','auxclick','contextmenu','scroll']
+        endEvents.forEach(e => document.addEventListener(e,endBrushing,false));
     }
 
-    createDivElem(width, height) {
-        const divElem = document.createElementNS(null, "div");
-        divElem.setAttributeNS(null, "class", "vis-scatterplot");
-        divElem.setAttributeNS(null, "viewBox", 0 + " " + 0 + " " + width + " " + height);
-        divElem.setAttributeNS(null, "width", "100%");
-        divElem.setAttributeNS(null, "height", "100%");
-        divElem.setAttributeNS(null, "transform", "matrix(1 0 0 -1 0 0)");
+    /**
+     * Helper function for zooming in after the scale has been updated.
+     */
+    zoomingHelper(scaleAndAxis,box_width,scatter,points) {
+        scaleAndAxis.xAxis.transition().duration(animation_duration)
+            .call(d3.axisBottom(scaleAndAxis.xScale).ticks(box_width / x_axis_label_width));
+        scaleAndAxis.yAxis.transition().duration(animation_duration)
+            .call(d3.axisLeft(scaleAndAxis.yScale));
+
+        scatter.selectAll("path")
+            .transition().duration(animation_duration)
+            .attr('transform',d => "translate(" + scaleAndAxis.xScale(d.x) + "," + scaleAndAxis.yScale(d.y) + ")")
+
+        if (points.labels === visilbe_points) {
+            scatter.selectAll("text")
+                .transition().duration(animation_duration)
+                .attr("x",d => scaleAndAxis.xScale(d.x) + point_label_padding_x)
+                .attr("y",d => scaleAndAxis.yScale(d.y) + point_label_padding_y)
+        }
+    }
+
+    /**
+     * Creates a plot object and populates it with given data.
+     */
+    createScatter(svg,box_width,box_height,points,dataPoints,scaleAndAxis) {
+        let clip = svg.append("defs").append("svg:clipPath")
+            .attr("id","clip")
+            .append("svg:rect")
+            .attr("width",box_width)
+            .attr("height",box_height)
+            .attr("x",0)
+            .attr("y",0);
+
+        let symbol = d3.symbol();
+
+        let scatter = svg.append('g')
+            .attr("clip-path","url(#clip)")
+
+        let sizeScaleMultiplier = 100
+
```

Comment:
You put enters inside of the body, which means the body should either be refactored to separate functions or sections should be added here. I understand why you added newlines - these things are really long here. And a section / refactoring would help a lot here. When reading the code its hard for me to understand what then ext 10 lines do - just add section with explanation! Every time you need to put enter inside of a body, you need to use sections or refactoring.